staging/src/k8s.io/apiserver/pkg/util/webhook/webhook.go

```diff
@@ -90,6 +90,11 @@ func WithExponentialBackoff(initialBackoff time.Duration, webhookFn func() error
 	var err error
 	wait.ExponentialBackoff(backoff, func() (bool, error) {
 		err = webhookFn()
+		// these errors indicate a need to retry an authentication check
+		if apierrors.IsServerTimeout(err) || apierrors.IsTimeout(err) || apierrors.IsTooManyRequests(err) {
```

Comment:
Not necessarily. We don't actually honor the delay right now, but we have two classes of things:

1. errors that mean retry
2. an error or response (that doesn't automatically mean retry) that still asks you to retry for some other reason